README.md

```diff
@@ -1,2 +1,270 @@
+|notificationToken|string|-|The FCM token as received from Firebase in the app. TBD: We may want to make this non-optional and simply not store device information, if there is no available token (e.g. if notifications are restricted).|
+|platform|optional string|e.g. "iOS", "Android"|This information is important as context for the `osVersion`, `appVersion` and `appBuild` properties.|
+|osVersion|optional string|e.g. "17.5.1"|The version of the OS. Depending on the OS, they may have different formats to be specified separately.|
+|appVersion|optional string|e.g. "1.0.1"|The version of the app as it is specified on the App/Play Store.|
+|appBuild|optional string|e.g. "56"|The build version of the app as defined by the CI pipeline releasing the app.|
+|language|optional string|e.g. "en"|Following IETF BCP-47 / [FHIR ValueSet languages](https://hl7.org/fhir/R4B/valueset-languages.html)|
+
+Push notifications over APNS / FCM only contain text in a single language. For this, the device's `language` property shall be prioritized, falling back on the user's `language` property and using US-English if both are not present.
+
+#### Adding a new device or modifying information of an existing one
+
+A device can only be identified by its notification token. When updating a device's information, we therefore check if the notification token already exists in Firestore. If it already exists, we update all other existing fields' values - otherwise, we create a new device.
+
+A device may receive a different notification token at any time though. Therefore, we might create a new device, even though that device already exists in the table with a different (now inactive) notification token. Therefore, every time we send out a new notification and receive the information that a token is no longer active, we need to remove the device from this table.
+
+### /users/$userId$/messages/$messageId$
+
+|Property|Type|Values|Comments|
+|-|-|-|-|
+|dueDate|optional DateTime|-|The due date of the message to be shown in-app.|
+|triggerDate|optional DateTime|-|A date to trigger a notification on the device. If not set, then there is no notification to trigger. The notification should be triggered by the server. TBD: Are there cases where a message would trigger multiple times or event- rather than date-driven?|
+|completionDate|optional DateTime|-|Specifies when a message has been completed. TBD: Messages containing a completionDate may either be hidden on user's devices or be shown crossed out.|
+|type|optional string|e.g. "questionnaireReminder"|Some messages are sent out on a regular basis, where only the most recent message is really relevant for the patient (e.g. a reminder for a questionnaire). With this property, we can easily find existing messages of the same type and replace them with a new one, if necessary.|
```

Comment:
Would be great of we can document all message types here and maybe take some of the information from the UI document in here to document some of the behavior here and persist it in the repo.

Reply:
I changed the action section to be a "Message types" section containing both formats for the actions and the available types.